src/grammar.lalrpop

```diff
@@ -561,6 +562,45 @@ ArgNames: ArgNames = {
     <label:VarName> <name:DiscardName> => ArgNames::LabelledDiscard { label, name },
 }
 
+RecordUpdate: UntypedExpr = {
+    // <s:@L> <m:VarName> "#" <n:UpName> <args:RecordUpdateArgs> <e:@L> => UntypedExpr::RecordUpdate {
+    //     location: location(s, e),
+    //     module: Some(m),
+    //     name: n,
+    //     spread: args.0,
+    //     args: args.1,
+    // },
+
+    <s:@L> "#" <n:UpName> <args:RecordUpdateArgs> <e:@L> => UntypedExpr::RecordUpdate {
+        location: location(s, e),
+        module: None,
+        name: n,
+        spread: args.0,
+        args: args.1,
+    }
+}
+
+RecordUpdateArgs: (RecordUpdateSpread, Vec<UntypedRecordUpdateArg<UntypedExpr>>) = {
+    "(" <spread:RecordUpdateSpread> <args:Comma<RecordUpdateArg>> ")" => {
+        (spread, args)
+    }
+}
+
+RecordUpdateSpread: RecordUpdateSpread = {
+    <s:@L> ".." <name:VarName> ","? <e:@L> => RecordUpdateSpread {
```

Comment:
Are these the best spots for the location tokens? I wonder if this rule would be more accurate:

```
 ".." <s:@L> <name:VarName> <e:@L> ","?
```

Reply:
The optional comma here is a little tricky, because it allows code like this:

```
let p = Person(..person age: 20)
```

I have an idea for resolving that though.